drivers/ethernet/eth_nxp_enet.c

```diff
@@ -1047,4 +1047,29 @@ static const struct ethernet_api api_funcs = {
 					CONFIG_ETH_INIT_PRIORITY,			\
 					&api_funcs, NET_ETH_MTU);
 
+DT_INST_FOREACH_STATUS_OKAY(NXP_ENET_MAC_INIT)
+
+
+/*
+ * ENET module-level management
+ */
+
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT nxp_enet
+
+#define NXP_ENET_INIT(n)								\
+											\
+int nxp_enet_##n##_init(void)								\
+{											\
+	clock_control_on(DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),				\
+			(void *)DT_INST_CLOCKS_CELL_BY_IDX(n, 0, name));		\
+											\
+	ENET_Reset((ENET_Type *)DT_INST_REG_ADDR(n));					\
+											\
+	return 0;									\
+}											\
+											\
+	/* Init the module before any of the MAC, MDIO, or PTP clock */			\
+	SYS_INIT(nxp_enet_##n##_init, POST_KERNEL, 0);
```

Comment:
Any reason this is tabbed in? Since it is not part of the function maybe the tab is not needed.

Reply:
it's part of the macro function